tests/main/layout-change/task.yaml

```
@@ -11,10 +11,24 @@ details: |
 execute: |
     echo Testing layout change
     echo "Removing snap"
-    sudo snap remove --purge layouttest || true
+    sudo snap remove --purge test-snapd-layout-change || true
 
     echo "Installing snap 1"
     sudo snap install --beta test-snapd-layout-change
 
     echo "Installing snap 2"
     sudo snap refresh --edge test-snapd-layout-change
+
+    echo Testing layout change with a daemon
+    echo "Removing snap"
+    sudo snap remove --purge test-snapd-layout-change-with-daemon || true
+
+    echo "Installing snap 1 with daemon"
+    sudo snap install --beta test-snapd-layout-change-with-daemon
+
+    echo "Installing snap 2 with daemon (this one must fail)"
+    if sudo snap refresh --edge test-snapd-layout-change-with-daemon ; then
+        false
+    else
+        true
+    fi
```

This is a nicer way to do this. I believe we can also use `not <command>` as a one-liner, but echo some test in these cases are nice

```suggestion
    if sudo snap refresh --edge test-snapd-layout-change-with-daemon ; then
        echo "running snap refresh for test-snapd-layout-change-with-daemon did not fail as expected"
        exit 1
    fi
```